servicetalk-encoding-netty/src/main/java/io/servicetalk/encoding/netty/DefaultContentCodecBuilder.java

```diff
@@ -0,0 +1,114 @@
+
+    protected int maxAllowedPayloadSize() {
+        return maxAllowedPayloadSize;
+    }
+
+    @Override
+    public ContentCodecBuilder maxAllowedPayloadSize(final int maxAllowedPayloadSize) {
+        if (maxAllowedPayloadSize <= 0) {
+            throw new IllegalArgumentException("maxAllowedPayloadSize: " + maxAllowedPayloadSize + " (expected > 0)");
+        }
+
+        this.maxAllowedPayloadSize = maxAllowedPayloadSize;
+        return this;
+    }
+
+    public abstract static class ZipContentCodecBuilder extends DefaultContentCodecBuilder {
+
+        private int compressionLevel = 6;
+
+        /**
+         * Sets the compression level for this codec's encoder.
+         * @param compressionLevel 1 yields the fastest compression and 9 yields the best compression.
+         * 0 means no compression. The default compression level is 6
+         * @return <code>this</code>
+         */
+        public ZipContentCodecBuilder withCompressionLevel(final int compressionLevel) {
+            this.compressionLevel = compressionLevel;
+            return this;
+        }
+
+        protected int compressionLevel() {
+            return compressionLevel;
+        }
+    }
+
+    static final class GzipContentCodecBuilder extends ZipContentCodecBuilder {
+        @Override
+        public ContentCodec build() {
+            return new NettyChannelContentCodec("gzip",
+                    () -> new JdkZlibEncoder(ZlibWrapper.GZIP, compressionLevel()),
+                    () -> new JdkZlibDecoder(ZlibWrapper.GZIP, maxAllowedPayloadSize())
+            );
+        }
+    }
+
+    static final class DeflateContentCodecBuilder extends ZipContentCodecBuilder {
+        @Override
+        public ContentCodec build() {
+            return new NettyChannelContentCodec("deflate",
+                    () -> new JdkZlibEncoder(ZlibWrapper.ZLIB, compressionLevel()),
+                    () -> new JdkZlibDecoder(ZlibWrapper.ZLIB, maxAllowedPayloadSize())
+            );
+        }
+    }
+
+    public static final class SnappyContentCodecBuilder extends DefaultContentCodecBuilder {
+
+        private boolean validateChecksums;
+
+        /**
+         * Force snappy to validate checksums, <code>false</code> by default.
+         * @param validateChecksums If true, the checksum field will be validated against the actual uncompressed data,
+         * and if the checksums do not match, a suitable DecompressionException will be thrown
+         * @return <code>this</code>
+         */
+        public SnappyContentCodecBuilder validateChecksums(boolean validateChecksums) {
+            this.validateChecksums = validateChecksums;
+            return this;
+        }
+
+        @Override
+        protected int maxAllowedPayloadSize() {
+            throw new UnsupportedOperationException("Max allowed payload size is not supported for snappy. " +
+                    "The maximum chunk size of snappy is " + Short.MAX_VALUE);
```

Comment:
Snappy 1.1 allows 64kb. I don't see it mentioned which version Netty supports.

Reply:
Good catch. I was referring to Netty's internal max encoding chunk size (https://github.com/netty/netty/blob/4.1/codec/src/main/java/io/netty/handler/codec/compression/SnappyFrameEncoder.java#L70) but its irrelevant to decompression. I will remove that sentence, we shouldn't rely on internals anyway.